### ISSUE TITLE
Dense Output Rodas5

### ISSUE THREAD

**Issue by gstein3m** [SciML/DifferentialEquations.jl#838]
The following program shows that dense output of Rodas5 is wrong for the algebraic components of DAE-problems:

using DifferentialEquations
using Plots

function dae!(dy, y, p, t)
    dy[1] = 3*t^2
    dy[2] = y[1] - y[2]
    nothing
end

#-- test dense output --
M = zeros(2,2); M[1,1] = 1.0
f = ODEFunction(dae!, mass_matrix = M);
y0 = [0.0; 0.0]; tspan = [0.0, 1.0];
prob = ODEProblem(f, y0, tspan, [])
sol = solve(prob, Rodas5())
println(sol.destats);
t = range(0.0,1.0,step = 0.1); 
scatter(t,t.^3, label = "exact sol")
plot!(sol)

I propose to implement dense output according to Rodas4/Rodas4p/Rodas4P2 with coefficients

h21= convert(T,31.040278745555575)
h22= convert(T,-6.925207756233034) 
h23= convert(T,41.54985263353498) 
h24= convert(T,-4.19983367949553) 
h25= convert(T,-20.0116097917112) 
h26= convert(T,-0.975683545855504) 
h27= convert(T,-1.0348884896058437) 

h31= convert(T,-1.3755295424732132) 
h32= convert(T,-1.1999290450148692e-12)
h33= convert(T,15.641155546064695) 
h34= convert(T,23.321973579898177) 
h35= convert(T,7.39545827189598)
h36= convert(T,-4.317455229088304) 
h37= convert(T,-3.409024912892793) 

Best regards 

Gerd

**Comment by ChrisRackauckas** [comment]
Yes, the documentation says this is the case and recommends not using the Rodas5 algorithm's dense output on DAEs.

>Rodas5 - A 5th order A-stable stiffly stable Rosenbrock method. Currently has a Hermite interpolant because its stiff-aware 3rd order interpolant is not yet implemented. This means the interpolation is inaccurate on algebraic variables, meaning this algorithm should not be used with saveat or post-solution interpolation on DAEs.

It's inherent to the method as it does not have a stiff-aware 3rd order interpolation derived, so this'll close since there's no solution other than to say that it's not recommended for this purpose for this reason.

**Comment by ChrisRackauckas** [comment]
> I propose to implement dense output according to Rodas4/Rodas4p/Rodas4P2 with coefficients

How did you derive those coefficients?

**Comment by ChrisRackauckas** [comment]
Notice that the algorithm already has the tie ins for the stiffly accurate interpolation. Add them here:

https://github.com/SciML/OrdinaryDiffEq.jl/blob/master/src/tableaus/rosenbrock_tableaus.jl#L515

uncomment

https://github.com/SciML/OrdinaryDiffEq.jl/blob/master/src/perform_step/rosenbrock_perform_step.jl#L1877-L1881

etc. and then you add to the dispatches

https://github.com/SciML/OrdinaryDiffEq.jl/blob/master/src/dense/rosenbrock_interpolants.jl#L82-L141

and it will use them. So it's easy to implement if the coefficients exist. I just never figured out what they should be nor took the time to derive them.

**Comment by gstein3m** [comment]
I am currently working on an improvement of rodas5: a) towards a rodas5p and b)  A-stability of the embedded method which may affect the stepsize selection.
Thereby I computed the coefficients of the 3rd order interpolant of rodas5. The coefficients must fullfill some order conditions according to the book of Hairer/Wanner.
I will try to implement anything according to your comments.
Gerd
